wolfcrypt/src/sha512.c

```diff
@@ -458,6 +544,84 @@ int wc_InitSha512_ex(wc_Sha512* sha512, void* heap, int devId)
     return ret;
 }
 
+#if !defined(HAVE_FIPS) && !defined(HAVE_SELFTEST)
+#if !defined(WOLFSSL_NOSHA512_224)
+int wc_InitSha512_224_ex(wc_Sha512* sha512, void* heap, int devId)
+{
+    int ret = 0;
+
+    if (sha512 == NULL)
+        return BAD_FUNC_ARG;
+
+    sha512->heap = heap;
+#ifdef WOLFSSL_SMALL_STACK_CACHE
+    sha512->W = NULL;
+#endif
+#ifdef WOLF_CRYPTO_CB
+    sha512->devId = devId;
+    sha512->devCtx = NULL;
+#endif
+
+    ret = InitSha512_224(sha512);
+    if (ret != 0)
+        return ret;
+
+#if defined(USE_INTEL_SPEEDUP) && \
+    (defined(HAVE_INTEL_AVX1) || defined(HAVE_INTEL_AVX2))
+    Sha512_SetTransform();
+#endif
+
+#if defined(WOLFSSL_ASYNC_CRYPT) && defined(WC_ASYNC_ENABLE_SHA512)
+    ret = wolfAsync_DevCtxInit(&sha512->asyncDev,
+                        WOLFSSL_ASYNC_MARKER_SHA512, sha512->heap, devId);
+#else
+    (void)devId;
+#endif /* WOLFSSL_ASYNC_CRYPT */
+
+    return ret;
+}
+#endif /* !WOLFSSL_NOSHA512_224 */
+#endif /* !HAVE_FIPS && !HAVE_SELFTEST */
+
+#if !defined(HAVE_FIPS) && !defined(HAVE_SELFTEST)
+#if !defined(WOLFSSL_NOSHA512_256)
+int wc_InitSha512_256_ex(wc_Sha512* sha512, void* heap, int devId)
```

Comment:
Nearly identical to prior function, aside from `InitSha512_256`. Consider merging common code into another function and adding an argument to specify 224 or 256.

Reply:
Introduced the InitSha512_family, which has common code among the InitSha512_xxx_ex functions.